File: 백준/Silver/16953. A → B/A → B.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./example.txt";
let [goal, num] = fs.readFileSync(filePath).toString().split(" ").map(Number);

let count = 0;
while (goal < num) {
  if (num % 10 == 1) {
    num = parseInt(num / 10);
  } else if (num % 2 == 0) {
    num /= 2;
  } else {
    break;
  }
  count++;
}
console.log(num == goal ? count + 1 : -1);

File: 백준/Silver/16953. A → B/A → B.py
goal, num = map(int,input().split())

ans  = 1

while goal < num:
    if num % 10 == 1:
        num //= 10
    elif num % 2 == 0:
        num /= 2
    else:
        break
    ans += 1

print(ans if num == goal else -1)